contracts/meta-staking/src/contract.rs

```diff
@@ -174,32 +197,127 @@ mod execute {
         Ok(Response::default().add_message(msg))
     }
 
-    // TODO finish me
+    enum Method {
+        Add,
+        Sub,
+    }
+
+    fn update_delegations<'a>(
+        deps: DepsMut<'a>,
+        info: MessageInfo,
+        validator: &str,
+        amount: Uint128,
+        method: Method,
+    ) -> Result<DepsMut<'a>, ContractError> {
+        let action = |validator_info: Option<Uint128>| -> Result<_, ContractError> {
+            match method {
+                Method::Sub => {
+                    let val = validator_info.ok_or(ContractError::NoDelegationsForValidator {})?;
+                    val.checked_sub(amount)
+                        .map_err(|_| ContractError::InsufficientDelegation {})
+                }
+                Method::Add => Ok(validator_info.unwrap_or_default() + amount),
+            }
+        };
+
+        VALIDATORS_BY_CONSUMER.update(deps.storage, (&info.sender, validator), action)?;
```

Comment:
I'm wary of this.
In the code undelegate you load these already. Then possibly modify them.

When you call update, you load again.

I would just pass in the loaded values here, modify as needed and `save`. Not `load` in here.

Otherwise, someone might modify eg. validator in the calling function and surprised it doesn't reflect when calling this

Reply:
Yes, this function was made on my first attempt, and now I see its really not needed, I need to modify it.